### PR TITLE
ecmd-pdbg istep control update

### PR DESCRIPTION
**Pull request by lkammath**
1. Enabled isteps proc_clock_test , set_ref_clock, sbe_config_update
    2. Replaced CFAM based istep mode settings with istep_init

### CODE REVIEW

**Comment by lkammath** [comment]
Still testing changes

**Comment by lkammath** [comment]
Tested and working well. Open for review @ojayanth @thejta

**Comment by lkammath** [comment]
Had a discussion with @ojayanth @apopple before pushing this changes .. Problem is since ipl_init() and InitTarget() now calls pdbg_target_init() there is pdbg assert. So, we have to make use of one. This change handles that.

File: src/dll/edbgEcmdDll.C [hunk 1]
```diff
@@ -1685,54 +1689,8 @@ uint32_t dllQueryHostMemInfoRanges( const std::vector<ecmdChipTarget> & i_target
 
 #ifdef EDBG_ISTEP_CTRL_FUNCTIONS
 
-enum IplMode_t
```
Comment (apopple) [inline src/dll/edbgEcmdDll.C:1688]: These (and the rest of the changes from here) seem unrelated to the commit description. Are they required? If so please split these into a separate commit. Thanks!

Comment (lkammath) [reply src/dll/edbgEcmdDll.C:1688]: Updated commit message accordingly. Removing old code with ipl_init() coming to set the istep attribute

Comment (apopple) [reply src/dll/edbgEcmdDll.C:1688]: This was not resolved, if the code is required please split into a separate commit. Thanks.

File: src/dll/edbgEcmdDll.C [hunk 2]
```diff
@@ -661,7 +661,11 @@ uint32_t dllInitDll() {
     gEDBG_HOME = realPath;
   }
 
+#ifdef EDBG_ISTEP_CTRL_FUNCTIONS
```
Comment (thejta) [inline src/dll/edbgEcmdDll.C:664]: Please add a comment block in the code to explain why we need to have a completely different init path when istep control is enabled?
Also, is that actually when you intend?  Do you want someone doing a getscom to call ipl_init just because istep code is compiled into their executable?

Comment (lkammath) [reply src/dll/edbgEcmdDll.C:664]: Ok. Yes, its intended change for now. otherwise, there will be double initialization and pdbg asserts like explained in the commit message. I will add a note. please see the call flow of ipl_init() it already does pdbg_target_init

https://github.ibm.com/phal/ipl/blob/master/libipl/libipl.C#L108 -> https://github.ibm.com/phal/ipl/blob/master/libipl/libipl.C#L121 -> https://github.ibm.com/phal/libekb_p10/blob/master/libekb.C#L70

**Comment by lkammath** [comment]
I think I messed up with split commits.  But, still went ahead and added comment to make it more clear

File: src/dll/edbgEcmdDll.C (hunk 2 above)
Comment (apopple) [reply src/dll/edbgEcmdDll.C:667]: Couple of points:

This change isn't "for now" - Based on the design ipl_init() will always call pdbg_targets_init(). At the moment ipl_init() doesn't do much else so doing it for just a getscom should be ok.

> otherwise, there will be double initialization and pdbg asserts like explained in the commit message

I can't see that in the commit message. As you mentioned it looks like you've merged this commit with an unrelated one though, so perhaps that's why the commit message disappeared. You will need to resplit this one out, let me know if you need some help doing so.

Comment (thejta) [reply src/dll/edbgEcmdDll.C:667]: I also also have issue with this - we shouldn't have to be calling a function labeled `ipl_init` to be able to perform basic hardware operations (i.e. `getscom`).  It makes the code unclear and makes determining ownership of the edbg init path difficult.  It seems like this need to be refined.

Why can't edbg continue to init pdbg since it is the loader of pdbg?  Then, if further on in the edbg execution we know we are calling istep functions, those then do the remaining ipl_init?

What you want to change here is the very start of the edbg load and that needs to be as clear as possible to not have unintended side effects.  At a minimum there needs to be a much better commit message explaining all the considerations and angles that went into wanting such a change.

Comment (lkammath) [reply src/dll/edbgEcmdDll.C:667]: with the earlier approach, the istep execution is crashing because pdbg_target_init gets called twice (once in dllInit and second-time ipl_init() called as part of istep functions and because of that there is double initialization ). 

I tested this code . There is no issues with this approach

Comment (thejta) [reply src/dll/edbgEcmdDll.C:667]: Just because it works doesn't mean it is the right answer.  I still need answers to my questions and my comments addressed before I'll approve.

Comment (ojayanth) [reply src/dll/edbgEcmdDll.C:667]: Agreed the Jason. Looks like not a clean approch. considering use case of IPL time hw access use case. This code is going to fail any hw access during ipl time. like 
ipl_init(IPL_HOSTBOOT) updates istep_mode attribute to istep mode.  The design direction to restrict attribute updates in hostboot ipl window.  Currently not enabled in the code.

**Comment by lkammath** [comment]
This branch seems to be got deleted and showing up as  unknown repository. So, created a new PR #41
